src/views/InteractionDesigner.vue

```diff
@@ -146,6 +170,27 @@ export class InteractionDesigner extends mixins(Lang, Routes) {
   @builderNamespace.Getter isResourceViewerCanvasEnabled!: boolean
   @clipboardNamespace.Getter isSimulatorActive!: boolean
 
+  // ###### Validation API Watchers [
+  @Watch('activeFlow', {deep: true, immediate: true})
+  async onActiveFlowChanged(newFlow: IFlow): Promise<void> {
+    this.debounceFlowValidation({newFlow})
+  }
+
+  @Watch('blocksOnActiveFlowForWatcher', {deep: true, immediate: true})
+  async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]): Promise<void> {
+    this.debounceBlockValidation()
+  }
+
+  @Watch('activeFlow.resources', {deep: true, immediate: true})
+  async onResourcesOnActiveFlowChanged(newResources: IResources, oldResources: IResources): Promise<void> {
+    console.debug('watch/activeFlow.resources:', 'resources inside active flow have changed', `from ${this.mainComponent}.`, 'Validating ...')
+    await this.validate_resourcesOnSupportedValues({
+      resources: newResources,
+      supportedModes: this.activeFlow.supported_modes,
+      supportedLanguages: this.activeFlow.languages,
+    })
+  }
+
```

Comment:
Thanks @bulatgab for this initiative, I'd be happy to see how this would go in store ;)
Could you pls create a new ticket for that under the resource-view epic https://viamoinc.atlassian.net/browse/VMO-6150 ? (because a subtask in phase1 might not be appropriate, in case we cannot finish that new refractor in time)

Reply:
You're right, I shouldn't have dived into it right away. I'll create a new ticket for that.
Thanks for approving the suggested change 👍

Reply:
https://viamoinc.atlassian.net/browse/VMO-7115